cot/src/auth.rs

```diff
@@ -142,6 +157,54 @@ pub trait User {
     /// If this method returns `None`, the session hash is not checked.
     ///
     /// [`AnonymousUser`] always returns `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    ///
+    /// use cot::auth::{Password, SessionAuthHash, User, UserId};
+    /// use cot::config::SecretKey;
+    /// use hmac::{Hmac, KeyInit, Mac};
+    /// use sha2::Sha512;
+    ///
+    /// struct MyUser {
+    ///     id: i64,
+    ///     password: Password,
+    /// }
+    ///
+    /// type SessionAuthHmac = Hmac<Sha512>;
+    ///
+    /// impl User for MyUser {
+    ///     fn id(&self) -> Option<UserId> {
+    ///         Some(UserId::Int(self.id))
+    ///     }
+    ///
+    ///     fn username(&self) -> Option<Cow<'_, str>> {
+    ///         Some(Cow::from(format!("user{}", self.id)))
+    ///     }
+    ///
+    ///     fn is_active(&self) -> bool {
+    ///         true
+    ///     }
+    ///
+    ///     fn is_authenticated(&self) -> bool {
+    ///         true
+    ///     }
+    ///
+    ///     fn session_auth_hash(&self, secret_key: &SecretKey) -> Option<SessionAuthHash> {
+    ///         // thanks to this, the session hash is invalidated when the user changes their password
+    ///         // and the user is automatically logged out
+    ///
+    ///         let mut mac = <Hmac<Sha512>>::new_from_slice(secret_key.as_bytes())
```

Comment:
A bit easier to read and follow
```suggestion
    ///         let mut mac = SessionAuthHmac::new_from_slice(secret_key.as_bytes())
```